golem_messages/datastructures.py

```diff
@@ -68,13 +68,35 @@ def __setitem__(self, key, value):
         validator = getattr(
             self, 'validate_{}'.format(key), None
         )
-
         if callable(validator):
             validator(value=value)  # noqa pylint:disable=not-callable
 
         super().__setitem__(key, value)
 
 
+class ValidatingDependentDictItems(ValidatingDict):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.validate_dependent_dict_items()
+
+    def validate_dependent_dict_items(self):
+        dependent_items_list = getattr(self, 'DEPENDENT_ITEMS', [])
+        for items in dependent_items_list:  # pylint: disable=not-an-iterable
+            validator_function_dict = dict()
+            assert len(items) == 2
```

Comment:
and again, apart from the above - why would we use a list of tuples and then assert that each entry is 2 items long instead of just using a dict for a mapping - which would be both a lot more straightforward and readable and would eliminate the need to use such artificial asserts? ...

Reply:
Yes, but you were talking about
```python
    TASK_TYPE_META_PARAMETERS = {
        TaskType.Blender.name: BlenderScriptPackage,
        TaskType.LuxRender.name: LuxRenderScriptPackage,
    }
```
which I have already changed from list to dict.

But, this assert is for 
```python
    DEPENDENT_ITEMS = [
        ('task_type', 'meta_parameters'),
    ]
```
to be sure that method `validate_dependent_dict_items` has `DEPENDENT_ITEMS` as a list of two item's tuples.

To be honest I don't really know how you want here create dict instead of tuple. Can you specify it more please?

Reply:
Also this has been changed and does not exist in current version.